### ISSUE TITLE
Problem: No makefile recipe to test archivematica-common 

### ISSUE THREAD

**Issue by ross-spencer** [artefactual-labs/am#55]
We currently have tests for the Archivematica Dashboard, Server, and MCP Client. We don't test the archivematicaCommon functions via the docker-compose makefile. 

We need to include an recipe along the lines of:

`test-archivematica-common:
	docker-compose run --workdir /src/archivematicaCommon --rm --user=root --entrypoint=py.test archivematica-dashboard`

This takes quite a long time: 
```
real	11m17.324s
user	0m5.192s
sys	0m0.462s
```
And currently fails.

We can also run it via the other images:

* `archivematica-mcp-client` (passes)
```
real	9m42.609s
user	0m6.016s
sys	0m0.971s
```

* `archivematica-server` (multiple failures `tests/test_database_functions.py EEEEEEEEEEEE` and `tests/test_execute_functions.py F`)
```
real	0m51.356s
user	0m6.141s
sys	0m0.919s
```

**Failed tests:**
```
tests/test_execute_functions.py F

============================================================================== FAILURES ===============================================================================
_________________________________________________________________________ test_capture_output _________________________________________________________________________

    def test_capture_output():
        """Tests behaviour of capture_output when executing sub processes."""
    
        # Test that stdout and stderr are not captured by default
        ret, std_out, std_err = execsub.launchSubProcess(['ls', '/tmp'])
        assert std_out is ''
        assert std_err is ''
    
        # Test that stdout and stderr are captured when `capture_output` is
        # enabled.
        ret, std_out, std_err = execsub.launchSubProcess(
            ['ls', '/tmp'], capture_output=True)
>       assert std_out is not '' or std_err is not ''
E       assert ('' is not '' or '' is not '')

tests/test_execute_functions.py:20: AssertionError
```
Example mcp-server failures:
```
request = <SubRequest '_django_setup_unittest' for <TestCaseFunction 'test_get_accession_number_from_transfer'>>
_django_cursor_wrapper = <pytest_django.plugin.CursorManager object at 0x7fe4c088b490>

    @pytest.fixture(autouse=True, scope='class')
    def _django_setup_unittest(request, _django_cursor_wrapper):
        """Setup a django unittest, internal to pytest-django."""
        if django_settings_is_configured() and is_django_unittest(request):
            request.getfuncargvalue('_django_test_environment')
            request.getfuncargvalue('_django_db_setup')
    
            _django_cursor_wrapper.enable()
    
            cls = request.node.cls
    
            _restore_class_methods(cls)
>           cls.setUpClass()

/usr/local/lib/python2.7/site-packages/pytest_django/plugin.py:386: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python2.7/site-packages/django/test/testcases.py:956: in setUpClass
    'database': db_name,
/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py:120: in call_command
    return command.execute(*args, **defaults)
/usr/local/lib/python2.7/site-packages/django/core/management/base.py:445: in execute
    output = self.handle(*args, **options)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:60: in handle
    self.loaddata(fixture_labels)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:100: in loaddata
    self.load_label(fixture_label)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:151: in load_label
    for obj in objects:
/usr/local/lib/python2.7/site-packages/django/core/serializers/json.py:85: in Deserializer
    six.reraise(DeserializationError, DeserializationError(e), sys.exc_info()[2])
/usr/local/lib/python2.7/site-packages/django/core/serializers/json.py:79: in Deserializer
    for obj in PythonDeserializer(objects, **options):
/usr/local/lib/python2.7/site-packages/django/core/serializers/python.py:96: in Deserializer
    Model = _get_model(d["model"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

model_identifier = 'main.agent'

    def _get_model(model_identifier):
        """
        Helper to look up a model from an "app_label.model_name" string.
        """
        try:
            return apps.get_model(model_identifier)
        except (LookupError, TypeError):
>           raise base.DeserializationError("Invalid model identifier: '%s'" % model_identifier)
E           DeserializationError: Problem installing fixture '/src/archivematicaCommon/tests/fixtures/agents.json': Invalid model identifier: 'main.agent'

/usr/local/lib/python2.7/site-packages/django/core/serializers/python.py:168: DeserializationError
________________________________________ ERROR at setup of TestDatabaseFunctions.test_get_acession_number_from_transfer_raises ________________________________________

request = <SubRequest '_django_setup_unittest' for <TestCaseFunction 'test_get_accession_number_from_transfer'>>
_django_cursor_wrapper = <pytest_django.plugin.CursorManager object at 0x7fe4c088b490>

    @pytest.fixture(autouse=True, scope='class')
    def _django_setup_unittest(request, _django_cursor_wrapper):
        """Setup a django unittest, internal to pytest-django."""
        if django_settings_is_configured() and is_django_unittest(request):
            request.getfuncargvalue('_django_test_environment')
            request.getfuncargvalue('_django_db_setup')
    
            _django_cursor_wrapper.enable()
    
            cls = request.node.cls
    
            _restore_class_methods(cls)
>           cls.setUpClass()

/usr/local/lib/python2.7/site-packages/pytest_django/plugin.py:386: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/usr/local/lib/python2.7/site-packages/django/test/testcases.py:956: in setUpClass
    'database': db_name,
/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py:120: in call_command
    return command.execute(*args, **defaults)
/usr/local/lib/python2.7/site-packages/django/core/management/base.py:445: in execute
    output = self.handle(*args, **options)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:60: in handle
    self.loaddata(fixture_labels)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:100: in loaddata
    self.load_label(fixture_label)
/usr/local/lib/python2.7/site-packages/django/core/management/commands/loaddata.py:151: in load_label
    for obj in objects:
/usr/local/lib/python2.7/site-packages/django/core/serializers/json.py:85: in Deserializer
    six.reraise(DeserializationError, DeserializationError(e), sys.exc_info()[2])
/usr/local/lib/python2.7/site-packages/django/core/serializers/json.py:79: in Deserializer
    for obj in PythonDeserializer(objects, **options):
/usr/local/lib/python2.7/site-packages/django/core/serializers/python.py:96: in Deserializer
    Model = _get_model(d["model"])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

model_identifier = 'main.agent'

    def _get_model(model_identifier):
        """
        Helper to look up a model from an "app_label.model_name" string.
        """
        try:
            return apps.get_model(model_identifier)
        except (LookupError, TypeError):
>           raise base.DeserializationError("Invalid model identifier: '%s'" % model_identifier)
E           DeserializationError: Problem installing fixture '/src/archivematicaCommon/tests/fixtures/agents.json': Invalid model identifier: 'main.agent'

/usr/local/lib/python2.7/site-packages/django/core/serializers/python.py:168: DeserializationError
```

**Comment by ross-spencer** [comment]
N.b. Some tests reference localhost, and won't resolve, e.g. `127.0.0.1:9200` for elasticsearch. This can be corrected by using the DNS in the Docker Environment, e.g. `elasticsearch:9200`. The `vcrpy` tapes will need re-recording at this point.

**Comment by ross-spencer** [comment]
**NB.** See here for information on when the tests are running too slow like above: https://github.com/artefactual-labs/am/tree/master/compose#tests-are-too-slow

**Comment by sevein** [comment]
Fixed by #84.
